sigopt/hyperopt/base.py

```diff
@@ -38,26 +41,28 @@ def upload(self, trials=None):
     self.uploaded_tids.update(ids)
     return ids
 
+  def trial_to_run(self, trial):
+    metadata = {'optimizer': 'hyperopt'}
+    result = trial['result']
+    metrics = {k:v for k, v in result.items() if isinstance(v, (int, float))}
+    parameters = self.trial_parameters(trial)
+    status = result.get('status')
+
+    run = RunBuilder(name=get_default_name(self.factory.project), metadata=metadata)
+    self.log_run_params(run, parameters)
```

Comment:
I think this should also be in the `run` object and not in `SigOptTrials`.

Reply:
yeah but now it's really hard to remember what is in the "builder" and what is on the run object. It makes more sense to have everything on `run.method()` and not some parts on `self.` and other `run.`

Reply:
ok. just change `RunBuilder` to `DictRunContext`, make a `HyperOptRunContext(DictRunContext)` class that only implements `log_run_params`

Reply:
Change to `DictRunContext`, add new function `log_parameters`, no need to create `HyperOptRunContext`

Reply:
agreed! thanks